### PR TITLE
Add support for provisioned aws_dynamodb_table

### LINKED ISSUE #28: Add prices for provisioned DynamoDB
DyanamoDB has two pricing options: privisioned and on-demand.
We will support the provisioned mode for starters.

### PR DESCRIPTION
**Pull request by sinabakh**
**Objective:** Add support for aws_rds_cluster terraform resource type
Status:

- [x] resource file
- [x] resource integration test
- [x] parsers
- [x] examples
- [x]  supported resources in README

**Issues:**
- Fixes #28 

Useful links:
- Overview: https://aws.amazon.com/dynamodb/
- Pricing: https://aws.amazon.com/dynamodb/pricing/provisioned/
- Terraform: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/dynamodb_table

### CODE REVIEW

**Comment by sinabakh** [comment]
Here is a sample table output for DynamoDB:
```
  NAME                                                             QUANTITY  UNIT  HOURLY COST  MONTHLY COST  

  aws_dynamodb_table.basic-dynamodb-table                                          
  ├─ Read capacity unit (RCU)                                         21900  hour       0.0039        2.8470  
  ├─ Write capacity unit (WCU)                                        14600  hour       0.0130        9.4900  
  ├─ global_table.us-east-2 Replicated write capacity unit (rWCU)     14600  hour       0.0195       14.2350  
  └─ global_table.us-west-1 Replicated write capacity unit (rWCU)     14600  hour       0.0217       15.8775  
  Total                                                                                 0.0582       42.4495  
                                                                                   
  OVERALL TOTAL                                                                         0.0582       42.4495
```

Currently, we can't have the support for following params:
- DataStorage
- Backup & Restore
- Streams
- Data Transfer

File: internal/terraform/aws/dyanmodb_table.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+	}
+	return quantity
+}
+
+func NewDynamoDBTable(address string, region string, rawValues map[string]interface{}) resource.Resource {
+	r := resource.NewBaseResource(address, rawValues, true)
+
+	// Check billing mode
+	billingMode := rawValues["billing_mode"]
+	if billingMode != "PROVISIONED" {
```
Comment (sinabakh) [inline internal/terraform/aws/dyanmodb_table.go:59]: @aliscott I couldn't figure out a suitable way to test this case in the integration test. Any help will be appreciated.

File: pkg/resource/resource.go [hunk 2]
```diff
@@ -32,7 +32,7 @@ type PriceFilter struct {
 	PurchaseOption     *string `json:"purchaseOption,omitempty"`
 	Unit               *string `json:"unit,omitempty"`
 	Description        *string `json:"description,omitempty"`
-	DescriptionRegex   *string `json:"descriptionRegex,omitempty"`
+	DescriptionRegex   *string `json:"description_regex,omitempty"`
```
Comment (sinabakh) [inline pkg/resource/resource.go:35]: It is a bug and has no direct relation with the main issue.

File: internal/terraform/aws/dyanmodb_table.go (hunk 1 above)
Comment (aliscott) [reply internal/terraform/aws/dyanmodb_table.go:59]: Yeah I was thinking about this as well for adding functionality to ensure all integrations tests log no warnings or errors. It seems logrus has some hooks we might be able to use: https://github.com/sirupsen/logrus#testing. I wouldn't block this PR on this though. I'll create a separate issue and add this to the list of tests to include in it.

File: internal/terraform/aws/dyanmodb_table.go [hunk 3]
```diff
@@ -0,0 +1,114 @@
+package aws
```
Comment (aliscott) [inline internal/terraform/aws/dyanmodb_table.go:1]: There's a typo in the filename here `dyanmodb_table.go` -> `dynamodb_table.go`

**Review by aliscott — CHANGES_REQUESTED** [review]

File: internal/terraform/aws/dyanmodb_table.go (hunk 3 above)
Comment (sinabakh) [reply internal/terraform/aws/dyanmodb_table.go]: Fixed

**Comment by sinabakh** [comment]
I think this PR is ready for merge.

File: README.md [hunk 4]
```diff
@@ -19,6 +19,9 @@ Currently this supports the following on-Demand and Spot pricing for the followi
  * `aws_lb`
  * `aws_nat_gateway`
 
+ It also supports provisioned pricing for the following ASW resources:
```
Comment (alikhajeh1) [inline README.md:22]: ASW -> AWS?

**Pull request merged**